src/main/kotlin/day18/repository/photo/PhotoRepositoryImpl.kt

```
@@ -0,0 +1,28 @@
+import io.ktor.client.plugins.*
+import kotlinx.io.IOException
+import org.example.day18.data_source.photo.PhotoDataSource
+import org.example.day18.mapper.photo.toPhoto
+import org.example.day18.model.photo.Photo
+import org.example.day18.util.photo.PhotoError
+import org.example.day18.util.photo.Result
+
+class PhotoRepositoryImpl(private val dataSource: PhotoDataSource) : PhotoRepository {
+    override suspend fun getPhotos(query: String): Result<List<Photo>, PhotoError> {
+        try {
+            if (query == "") throw IllegalArgumentException()
```

detekt (1.23.7)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)